enterprise/workers/socket/src/durable-objects/websocket-room.ts

```diff
@@ -276,4 +292,146 @@ export class WebSocketRoom extends DurableObject<IEnv> {
       );
     }
   }
+
+  private extractContextKeysFromHeader(request: Request): string[] | undefined {
+    const contextKeysHeader = request.headers.get('X-Context-Keys');
+
+    if (!contextKeysHeader) {
+      return undefined;
+    }
+
+    try {
+      return JSON.parse(contextKeysHeader);
+    } catch (e) {
+      console.error('Failed to parse contextKeys:', e);
+
+      return undefined;
+    }
+  }
+
+  /**
+   * Log connection acceptance with context information
+   */
+  private logConnectionAccepted(userId: string, contextKeys?: string[]): void {
+    const contextDisplay = this.formatContextDisplay(contextKeys);
+    console.log(`Connection accepted for ${userId} with contexts: ${contextDisplay}`);
```

Comment:
We can skip this one, as logs are quite pricey, unless you want to check some specific things